src/ScalarPotentials/ImpurityScale.jl

```diff
@@ -46,4 +46,23 @@ function ImpurityScale(nt::NamedTuple)
     N = get_number_of_dimensions(grid)
     ImpurityScale{T, N, S, typeof(grid.axes)}(nt.values, grid)
 end
-Base.convert(T::Type{ImpurityScale}, x::NamedTuple) = T(x)
\ No newline at end of file
+Base.convert(T::Type{ImpurityScale}, x::NamedTuple) = T(x)
+
+
+"""
+    get_ticks_at_positions_of_edge_of_depleted_volumes(impscale::ImpurityScale{T, 3})
+
+The impurity scale field is analyzed in order to find and return ticks where the gradient is strong,
+which is the case at the surface of the depleted volume of a semiconductor.
+"""
+function get_ticks_at_positions_of_edge_of_depleted_volumes(impscale::ImpurityScale{T, 3}) where T
+    dims = (1, 2, 3)
+    mps::NTuple{3, Vector{T}} = broadcast(idim -> StatsBase.midpoints(impscale.grid[idim]), dims)
+    extended = length.(mps) .> 0
+    max_diffs = broadcast(idim -> extended[idim] ? map(i->maximum(abs.(selectdim(impscale.data, idim, i+1) .- selectdim(impscale.data, idim, i))), eachindex(mps[idim]))::Vector{T} : T[], dims);
+    inds = broadcast(idim -> extended[idim] ? findall(Δ -> 0.5 < Δ < 1, max_diffs[idim]) : Int[], dims) # 0.5 seems to be a good limit as we don't want to add too many ticks. 
```

Comment:
Not a fan of creating empty arrays here for dimensions in which the grid is not extended..